### PR TITLE
ENH: Add Dockerfile for development inside a container

### PR DESCRIPTION
**Pull request by ColCarroll**
This is tweaking #1473 -- test suite works inside the container, and running `. ./scripts/start_container.sh` works on my machine.  

It is ready to merge -- it does not change existing functionality (or add new features -- it is only for ease of development), but I can start a new 3.1 branch with this if we want to be strict about only bug fixes going into 3.0 @fonnesbeck.

### CODE REVIEW

**Comment by springcoil** [comment]
I think this is fine for 3.0

**Comment by AustinRochford** [comment]
Thanks for figuring this out @ColCarroll!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -104,9 +105,27 @@ tools:
   $ autopep8 path/to/pep8.py
   ```
 
+## Developing in Docker
+
+We have provided a Dockerfile which helps for isolating build problems, and local development.
+Install [Docker](https://www.docker.com/) for your operating system, clone this repo, then
+then run `./scripts/start_container.sh`. This should start a local docker container called `pymc3`,
```
Comment (twiecki) [inline CONTRIBUTING.md]: typo then then

**Comment by twiecki** [comment]
Since this is just adding new code I think we should merge this into master after the typo is fixed.

**Comment by twiecki** [comment]
I added an auto-build to docker hub: https://hub.docker.com/r/pymc/pymc3/ (let me know if I should add you to the docker org).

So you don't have to build the image anymore, you can just do: `docker pull pymc/pymc3`

**Comment by twiecki** [comment]
`docker run -d -p 8888:8888 pymc/pymc3`

Brings up the jupyter notebook, but you can't import pymc3. I now realize that this docker image isn't meant to be used by a user, although we could easily extend/change it to do that.

**Comment by ColCarroll** [comment]
@twiecki interesting idea.  what about building pymc in docker properly (i.e., pip installing it), and having a standalone `pymc3-docker` project?  Then the `start-container.sh` file could still mount the local pymc and edit the `$PYTHONPATH`.  

My worry is that since it is pretty easy to install, the docker image wouldn't be used, so there isn't much upside.

**Comment by twiecki** [comment]
Yeah, good point. Since we have it working on conda and linux is easy to begin with I agree with your assessment. In any case, we can use the dockerhub for easier debugging of travis problems now.

**Pull request merged**